File: backend/types/article.go
package types

import "time"

type Article struct {
	ID             string    `json:"id"`
	Slug           string    `json:"slug"`
	Title          string    `json:"title"`
	Content        string    `json:"content"`
	ContentPreview string    `json:"contentPreview"`
	CoverUrl       string    `json:"coverUrl"`
	PublishDate    time.Time `json:"publishDate"`
	Publisher      User      `json:"publisher"`
	Tags           []*Tag    `json:"tags"`
}

File: web/types/article.ts
import type { ITag } from "~/types/tag";
import type { IUser } from "~/types/user";

export interface IArticle {
  readonly id: string;
  readonly title: string;
  readonly slug: string;
  readonly contentPreview: string;
  readonly content: string;
  readonly coverUrl: string;
  readonly publishDate: string;
  readonly publisher: IUser;
  readonly tags: ITag[];
}
